src/test/providers/terminal.unit.test.ts

```diff
@@ -172,33 +177,70 @@ suite('Terminal Provider', () => {
             serviceContainer.setup(c => c.get(TypeMoq.It.isValue(ITerminalActivator))).returns(() => terminalActivator.object);
 
             terminal = TypeMoq.Mock.ofType<Terminal>();
+            getActiveResource.callsFake(() => resource);
+        });
+
+        teardown(() => {
+            sinon.restore();
         });
 
         test('If terminal.activateCurrentTerminal setting is set, provided terminal should be activated', async () => {
             terminalSettings.setup(t => t.activateEnvInCurrentTerminal).returns(() => true);
+            configService
+                .setup(c => c.getSettings(resource))
+                .returns(() => pythonSettings.object)
+                .verifiable(TypeMoq.Times.once());
 
             terminalProvider = new TerminalProvider(serviceContainer.object);
             await terminalProvider.initialize(terminal.object);
 
             terminalActivator.verify(a => a.activateEnvironmentInTerminal(terminal.object, undefined, true), TypeMoq.Times.once());
+            assert.ok(getActiveResource.calledOnce);
+            configService.verifyAll();
         });
 
         test('If terminal.activateCurrentTerminal setting is not set, provided terminal should not be activated', async () => {
             terminalSettings.setup(t => t.activateEnvInCurrentTerminal).returns(() => false);
+            configService
+                .setup(c => c.getSettings(resource))
+                .returns(() => pythonSettings.object)
+                .verifiable(TypeMoq.Times.once());
 
             terminalProvider = new TerminalProvider(serviceContainer.object);
             await terminalProvider.initialize(terminal.object);
 
             terminalActivator.verify(a => a.activateEnvironmentInTerminal(TypeMoq.It.isAny(), undefined, true), TypeMoq.Times.never());
+            assert.ok(getActiveResource.calledOnce);
+            configService.verifyAll();
         });
 
         test('terminal.activateCurrentTerminal setting is set but provided terminal is undefined', async () => {
             terminalSettings.setup(t => t.activateEnvInCurrentTerminal).returns(() => true);
+            configService
+                .setup(c => c.getSettings(resource))
+                .returns(() => pythonSettings.object)
+                .verifiable(TypeMoq.Times.once());
 
             terminalProvider = new TerminalProvider(serviceContainer.object);
             await terminalProvider.initialize(undefined);
 
             terminalActivator.verify(a => a.activateEnvironmentInTerminal(TypeMoq.It.isAny(), undefined, true), TypeMoq.Times.never());
+            assert.ok(getActiveResource.calledOnce);
+            configService.verifyAll();
+        });
+
+        test('Exceptions are swallowed if initializing terminal provider fails', async () => {
+            terminalSettings.setup(t => t.activateEnvInCurrentTerminal).returns(() => true);
```

Comment:
Let's stay consistent when formatting.

```suggestion
            terminalSettings
                .setup(t => t.activateEnvInCurrentTerminal)
                .returns(() => true);
```